File: app/src/main/java/io/github/seniorzhai/architecturesample/ui/UserModule.java
package io.github.seniorzhai.architecturesample.ui;

import android.content.Context;
import android.location.LocationManager;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

/**
 * Created by zhai on 2017/10/17.
 */

@Module
public class UserModule {

    private final Context context;

    public UserModule(Context context) {
        this.context = context;
    }

    @Provides
    @Singleton
    LocationManager provideLocationManager() {
        return (LocationManager) context.getSystemService(Context.LOCATION_SERVICE);
    }
}

File: app/src/main/java/io/github/seniorzhai/architecturesample/ui/UserModule.kt
package io.github.seniorzhai.architecturesample.ui

import android.arch.lifecycle.ViewModelProviders
import android.support.v7.app.AppCompatActivity
import dagger.Module
import dagger.Provides
import io.github.seniorzhai.architecturesample.Injection
import io.github.seniorzhai.architecturesample.di.ActivityScope

@Module
class UserModule(private val context: AppCompatActivity) {
    @Provides
    @ActivityScope
    internal fun provideUserViewModel(): UserViewModel {
        return ViewModelProviders
                .of(context, Injection.provideViewModelFactory(context))
                .get(UserViewModel::class.java)
    }
}
